File: tyama_codeeval50.pl
#!/usr/bin/perl
use strict;
use warnings;
while(<>){
	chomp;
	my ($s,$t)=split(';');
	my @a=split(',',$t);
	for(my $i=0;$i<=$#a/2;$i++){$s=~s/$a[$i*2]/chr($i+65)/eg;}
	for(my $i=0;$i<=$#a/2;$i++){my $x=chr($i+65);$s=~s/$x/$a[$i*2+1]/eg;}
	print $s."\n";
}

File: codeeval/tyama_codeeval50.rb
#!/usr/bin/ruby
while gets
s,t=$_.chomp.split(';')
a=t.split(',').each_slice(2).to_a.transpose
a[0].each_with_index{|e,i|s.gsub!(e,(i+65).chr)}
a[1].each_with_index{|e,i|s.gsub!((i+65).chr,e)}
puts s
end